packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -2024,10 +2079,21 @@ export class SnapController extends BaseController<
     delete snap.blockInformation;
 
     // store the snap back in state
-    this.update((state: any) => {
+    const { inversePatches } = this.update((state: any) => {
       state.snaps[snapId] = snap;
     });
 
+    // checking for isUpdate here as this function is also used in
+    // the install flow, we do not care to create snapshots for installs
+    if (isUpdate) {
+      const rollbackSnapshot = this.#getRollbackSnapshot(
+        snapId,
+      ) as RollbackSnapshot;
```

Comment:
Why does this need to be cast? TypeScript should be able to infer the type with just the assertion below.